tests/e2e/scenarios/test_provider_fault_proxy.py

```
@@ -345,3 +346,106 @@ def test_every_published_profile_has_a_self_test():
+    await asyncio.sleep(0)
+
+    assert proxy.state["pending_faults"] == 0, proxy.state
+
+
+async def test_world_reset_covers_every_provider_not_just_the_first():
+    """One un-reset provider is enough to leak, so the loop must cover all."""
+    upstreams = []
+    proxies = {}
+    try:
+        for service in ("google", "slack", "github"):
+            url, requests, runner = await _start_upstream()
```

Ruff (0.15.21)

[warning] Unpacked variable `requests` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)